internal/c/parts/video/image/image.cpp

```diff
@@ -199,16 +242,24 @@ static uint32_t *image_svg_load_from_memory(const uint8_t *buffer, size_t size,
     if (!svgString)
         return nullptr;
 
-    memcpy(svgString, buffer, size);
-    svgString[size] = '\0';
+    // Bail if we have binary data. We'll also copy the data while doing the check to avoid another pass
+    for (size_t i = 0; i < size; i++) {
+        auto c = svgString[i] = buffer[i];
```

Comment:
Note that `char` is typically signed so the +127 unicode values would trigger the `c < 32` check (I think the `auto` will be `char` due to the type of `svgString`). You should probably specify unsigned here if we want to allow them.

Reply:
You are correct. Nice catch!